### PR TITLE
Dynamic connectivity problem

### PR DESCRIPTION
**Pull request by tatsiak**
Created method count, method component and tests for them.

### CODE REVIEW

File: app/unionFind.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        this._elements = [];
+    }
+
+    connect(a, b) {
+        if (this.isConnected(a, b)) return;
+        if (isNaN(this._elements[a])) {
+         this._elements[a] = a;
+             // this._elements[1]=1;
+         }
+         this._elements[b] = isNaN(this._elements[b]) ? b : this._elements[b];
```
Comment (pavlom0x) [inline app/unionFind.js]: it's better to choose one way to set elements
or `if` or `?`

File: app/unionFind.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+             }
+         }
+     }
+
+    isConnected(a, b) {
+         // [undefined,undefined]
+         if (isNaN(this._elements[a]) || isNaN(this._elements[b])) {
+             return false;
+         }
+        return this._elements[a] === this._elements[b];
```
Comment (pavlom0x) [inline app/unionFind.js]: here some indentation issue

File: app/unionFind.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     }
+
+    count() {
+         let unique = [];
+         for (let i = 0; i < this._elements.length; i++) {
+            if (unique.indexOf(this._elements[i]) == -1 && !isNaN(this._elements[i])) {
+             unique.push(this._elements[i]);
+         }
+     }
```
Comment (pavlom0x) [inline app/unionFind.js]: here some indentation issue

File: test/unionFind.spec.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+/**
+ * Created by tatsiak on 4/1/17.
+ */
+ 'use strict'
+ const algolib = require('../index.js');
+ 
+
```
Comment (pavlom0x) [inline test/unionFind.spec.js]: do you actually need this extra empty line?

File: test/unionFind.spec.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
+/**
+ * Created by tatsiak on 4/1/17.
+ */
+ 'use strict'
+ const algolib = require('../index.js');
+ 
+ 
+ describe.only('#unionFind', function () {
```
Comment (pavlom0x) [inline test/unionFind.spec.js]: forgot `only`
be careful, if you break something in the code, no one will see it

File: test/unionFind.spec.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+     // isConnected
+     it('should create proper data structure', function () {
+       let uf = new algolib.UnionFind();
+       uf.connect.should.be.instanceOf(Function);
+       uf.isConnected.should.be.instanceOf(Function);
+   })
+
```
Comment (pavlom0x) [inline test/unionFind.spec.js]: indentation in all test cases

File: test/unionFind.spec.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+     it('should return true if elements are connected', function () {
+       let uf = new algolib.UnionFind();
+       uf.connect(1, 5);
+       uf.connect(5, 10);
+       let isElementsConnected = uf.isConnected(1, 10);
+       isElementsConnected.should.be.true();
+   })
+
+     // count
+     it('should return true if number of components is 1', function () {
```
Comment (pavlom0x) [inline test/unionFind.spec.js]: it should return number of components not `true` or `false`

File: test/unionFind.spec.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+       uf.isConnected.should.be.instanceOf(Function);
+   })
+
+     it('should return false if elements are not connected', function () {         
+        let uf = new algolib.UnionFind();
+        let isElementsConnected = uf.isConnected(7, 3333);
+        isElementsConnected.should.be.false();
+    }) 
+     // connect
+     it('should return true if elements are connected', function () {
```
Comment (pavlom0x) [inline test/unionFind.spec.js]: you can make sub group of tests 
```
describe('#unionFind', function () {
    describe('#connect', function () {
        it(...);
    })
})
```

File: test/unionFind.spec.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+       componentName.should.be.eql(5);
+   })
+
+    it('should return undefined if element not exists', function () {
+       let uf = new algolib.UnionFind();
+       uf.connect(3, 8);
+       (uf.component(91) ===undefined).should.be.true();
+   })
+
+    it('should return component name', function () {
```
Comment (pavlom0x) [inline test/unionFind.spec.js]: you already have this test above

**Comment by pavlom0x** [comment]
You should start working with new feature from `master` not from your last branch

if you just finish work on the old task, then your next steps are:
```
git checkout master //first go to master
git pull origin master // update it
git checkout -b my-new-bright-future // create a new branch and goes here
```
now you can work with new one in new branch

**Review by pavlom0x — CHANGES_REQUESTED** [review]

File: app/unionFind.js (hunk 2 above)
Comment (tatsiak) [reply app/unionFind.js]: sorry, I've tried to reformat code in sublime.... Now I'm going to use PHPStorm

File: test/unionFind.spec.js (hunk 4 above)
Comment (tatsiak) [reply test/unionFind.spec.js]: fixed

File: app/unionFind.js (hunk 3 above)
Comment (tatsiak) [reply app/unionFind.js]: fixed

File: test/unionFind.spec.js (hunk 6 above)
Comment (tatsiak) [reply test/unionFind.spec.js]: fixed

File: test/unionFind.spec.js (hunk 8 above)
Comment (tatsiak) [reply test/unionFind.spec.js]: oh, great idea
implemented it

File: test/unionFind.spec.js (hunk 5 above)
Comment (tatsiak) [reply test/unionFind.spec.js]: ok, sure
it was used only for debug.

File: test/unionFind.spec.js (hunk 9 above)
Comment (tatsiak) [reply test/unionFind.spec.js]: second test is a bit more complicated
So i decided to left only the second one.

File: test/unionFind.spec.js (hunk 7 above)
Comment (tatsiak) [reply test/unionFind.spec.js]: fixed

File: app/unionFind.js (hunk 1 above)
Comment (tatsiak) [reply app/unionFind.js]: in this case it was useful to use a ternary operator
Is there some rules about using only "if else" or only ternary operators?

**Pull request merged**